R/variable-derivation.R

```diff
@@ -0,0 +1,124 @@
+#' Get or set a derived variable's `CrunchExpr`
+#'
+#' Get a derived variable's derivation formula as a (`CrunchExpr`)[expressions] with
+#' `derivation([variable])`. Set (change) a derived variable's derivation with
+#' `derivation([variable]) <- [expression]`.
+#'
+#' To integrate (aka realize or instantiate) a variable so that it is no longer
+#'  dependent on the variables the derivation relies on, use
+#'  `derivation([variable]) <- NULL`
+#'
+#' `is.derived` can be used to see if a variable is derived or not. Additionally
+#' setting a derived variable's `is.derived` to `FALSE` will integrate the
+#' derived variable.
+#'
+#' @param x a variable
+#' @param value a `CrunchExpr` to be used as the derivation (for the setter
+#' only) or `NULL` to integrate a derived variable. For `is.derived`, `FALSE`
+#' can be used to integrate a derived variable.
+#'
+#' @return a `CrunchExpr` of the derivation
```

Comment:
Update this to say what is.derived and is.derived<- return. Also, derivation<- doesn't return a CrunchExpr